sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -358,52 +381,62 @@ private async Task PopulatePulumiVersionAsync(CancellationToken cancellationToke
 
         internal static void ValidatePulumiVersion(SemVersion minVersion, SemVersion currentVersion)
         {
-            if (minVersion.Major < currentVersion.Major) {
+            if (minVersion.Major < currentVersion.Major)
+            {
                 throw new InvalidOperationException($"Major version mismatch. You are using Pulumi CLI version {currentVersion} with Automation SDK v{minVersion.Major}. Please update the SDK.");
             }
-            if (minVersion > currentVersion) {
+            if (minVersion > currentVersion)
+            {
                 throw new InvalidOperationException($"Minimum version requirement failed. The minimum CLI version requirement is {minVersion}, your current CLI version is {currentVersion}. Please update the Pulumi CLI.");
             }
         }
 
         /// <inheritdoc/>
         public override async Task<ProjectSettings?> GetProjectSettingsAsync(CancellationToken cancellationToken = default)
         {
-            foreach (var ext in SettingsExtensions)
+            var path = this.FindSettingsFile();
+            var isJson = Path.GetExtension(path) == ".json";
+            if (!File.Exists(path))
+            {
+                return null;
+            }
+            else
```

Comment:
I think this `else` can be removed to reduce nesting, since we already return if the file does not exist

Reply:
Ah right this is one of those style things i'm unsure about. I'll take your suggestion in this case. In F# there is high tolerance for nesting as indeed the language does not even have an early return construct :)